src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -410,24 +410,54 @@ def _query_workspace_with_load_id(
     workspace_id: int,
     load_id: str,
     obj_names: list[str],
-) -> tuple[list[str], list[str]]:
+    assembly_objs_only: bool = True,
+) -> tuple[list[str], list[str], list[Any], list[Any]]:
     if len(obj_names) > _WS_MAX_BATCH_SIZE:
         raise ValueError(
             f"The effective max batch size must be <= {_WS_MAX_BATCH_SIZE}"
         )
     refs = [{"wsid": workspace_id, "name": name} for name in obj_names]
-    res = ws.get_object_info3(
-        {"objects": refs, "ignoreErrors": 1, "includeMetadata": 1}
-    )
-    uploaded_obj_names_batch = [
-        info[1]
-        for info in res["infos"]
-        if info is not None and "load_id" in info[10] and info[10]["load_id"] == load_id
-    ]
-    uploaded_obj_upas_batch = [
-        path[0].replace("/", "_") for path in res["paths"] if path is not None
-    ]
-    return uploaded_obj_names_batch, uploaded_obj_upas_batch
+    res = ws.get_object_info3({"objects": refs, "ignoreErrors": 1, "includeMetadata": 1})
+    uploaded_objs_info = [info for info in res["infos"] if info and info[10].get("load_id") == load_id]
+    if not uploaded_objs_info:
+        return list(), list(), list(), list()
+    uploaded_obj_names_batch = [info[1] for info in uploaded_objs_info]
+    uploaded_obj_upas_batch = ["{6}_{0}_{4}".format(*info) for info in uploaded_objs_info]
```

Comment:
There's a couple more info -> upa implementations below and I'm sure there's more in other places around the the code base. We should just make a function in src/common like

```
def obj_info_to_upa(upa, underscore_sep=False) -> str:
    sep = "_" if underscore_sep else "/"
    return f"{info[6]}{sep}{info[0]}{sep}{info[4]}
```

...and start replacing the zillion independent implementations

Reply:
👍 updated a few places I found in the uploader and downloader. Will replace others as I encounter them.

Reply:
:+1: